### PR TITLE
MLHR-1721: Convert NullPointerException to IOException during seek to avoid operator failure.

### PR DESCRIPTION
**Pull request by tushargosavi**

### CODE REVIEW

File: contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java [hunk 1]
```diff
@@ -85,7 +85,11 @@ public void reset() throws IOException
   @Override
   public boolean seek(Slice key) throws IOException
   {
-    return scanner.seekTo(key.buffer, key.offset, key.length);
+    try {
+      return scanner.seekTo(key.buffer, key.offset, key.length);
+    } catch (NullPointerException ex) {
+      throw new IOException("Converting NPE to IOException");
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: Please check whether stream was closed an only then do this mapping.

Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: I have added closed flag to Reader and converting to IOException only when reader is closed.

File: contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java [hunk 2]
```diff
@@ -85,7 +89,14 @@ public void reset() throws IOException
   @Override
   public boolean seek(Slice key) throws IOException
   {
-    return scanner.seekTo(key.buffer, key.offset, key.length);
+    try {
+      return scanner.seekTo(key.buffer, key.offset, key.length);
+    } catch (NullPointerException ex) {
+      if (closed)
+        throw new IOException("Converting NPE to IOException");
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: Please use a proper message: "Stream was closed"

File: contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java [hunk 3]
```diff
@@ -39,12 +40,14 @@
...
   private final Scanner scanner;
   private final FSDataInputStream fsdis;
+  private boolean closed = false;
 
   public TFileReader(FSDataInputStream fsdis, long fileLength, Configuration conf) throws IOException
   {
     this.fsdis = fsdis;
     reader = new Reader(fsdis, fileLength, conf);
     scanner = reader.createScanner();
+    closed = false;
```
Comment (tweise) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: not needed

Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: removed this statement.

File: contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java (hunk 2 above)
Comment (tushargosavi) [inline contrib/src/main/java/com/datatorrent/contrib/hdht/tfile/TFileReader.java]: changed as per suggestion.

**Pull request merged**